src/nvim/eval/funcs.c

```diff
@@ -10761,7 +10761,7 @@ static void f_strcharpart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
       if (off < 0) {
         len += 1;
       } else {
-        len += (size_t)MB_CPTR2LEN((const char_u *)p + off);
+        len += (size_t)utf_ptr2len((const char_u *)p + off);
```

Comment:
Why is there cast here originally? both `utf_ptr2len` and `len` are of type `int`, so this is redundant and probably unsafe.